hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestStorageContainerManager.java

```diff
@@ -323,17 +323,7 @@ public void testBlockDeletionTransactions() throws Exception {
       // after sometime, all the TX should be proceed and by then
       // the number of containerBlocks of all known containers will be
       // empty again.
-      GenericTestUtils.waitFor(() -> {
-        try {
-          if (SCMHAUtils.isSCMHAEnabled(cluster.getConf())) {
-            cluster.getStorageContainerManager().getScmHAManager()
-                .asSCMHADBTransactionBuffer().flush();
-          }
-          return delLog.getNumOfValidTransactions() == 0;
-        } catch (IOException e) {
-          return false;
-        }
-      }, 1000, 22000);
+      OzoneTestUtils.waitBlockDeleted(cluster.getStorageContainerManager());
```

Comment:
hmm this isn't an exact replacement because OzoneTestUtils.waitBlockDeleted() doesn't flush SCMHADBTransactionBuffer

Reply:
This place is for waiting blocks to be deleted. The test flush SCMHADBTransactionBuffer while creating delete transaction log. See https://github.com/apache/ozone/blob/dcbc1365c67faf90d3b6f1dad63569042ebbe6d6/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/TestStorageContainerManager.java#L989-L997

Reply:
Ah ok